x/stake/queryable.go

```diff
@@ -0,0 +1,224 @@
+package stake
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	keep "github.com/cosmos/cosmos-sdk/x/stake/keeper"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// nolint
+const (
+	QueryValidators          = "validators"
+	QueryValidator           = "validator"
+	QueryDelegator           = "delegator"
+	QueryDelegation          = "delegation"
+	QueryUnbondingDelegation = "unbondingDelegation"
+	QueryDelegatorValidators = "delegatorValidators"
+	QueryDelegatorValidator  = "delegatorValidator"
+	QueryPool                = "pool"
+	QueryParameters          = "parameters"
+)
+
+// creates a querier for staking REST endpoints
+func NewQuerier(k keep.Keeper) sdk.Querier {
+	return func(ctx sdk.Context, path []string, req abci.RequestQuery) (res []byte, err sdk.Error) {
+		switch path[0] {
+		case QueryValidators:
+			return queryValidators(ctx, path[1:], k)
+		case QueryValidator:
+			return queryValidator(ctx, path[1:], req, k)
+		case QueryDelegator:
+			return queryDelegator(ctx, path[1:], req, k)
+		case QueryDelegation:
+			return queryDelegation(ctx, path[1:], req, k)
+		case QueryUnbondingDelegation:
+			return queryUnbondingDelegation(ctx, path[1:], req, k)
+		case QueryDelegatorValidators:
+			return queryDelegatorValidators(ctx, path[1:], req, k)
+		case QueryDelegatorValidator:
+			return queryDelegatorValidator(ctx, path[1:], req, k)
+		case QueryPool:
+			return queryPool(ctx, k)
+		case QueryParameters:
+			return queryParameters(ctx, k)
+		default:
+			return nil, sdk.ErrUnknownRequest("unknown stake query endpoint")
+		}
+	}
+}
+
+// defines the params for the following queries:
+// - 'custom/stake/delegator'
+// - 'custom/stake/delegatorValidators'
+type QueryDelegatorParams struct {
```

Comment:
Why do these need to be their own struct? Are they fulfilling some interface?

Reply:
Did you mean to add these into the `NewQuerier` function above?

Reply:
It's just following the same implementation that was introduced in the `gov` queriers